iidm/iidm-xml-converter/src/test/java/com/powsybl/iidm/xml/IdentifiableExtensionXmlSerializerTest.java

```diff
@@ -191,26 +194,48 @@ public void testTerminalExtension() throws IOException {
         assertSame(loadXml.getTerminal(), terminalMockExtXml.getTerminal());
 
         // backward compatibility 1.0
-        //roundTripVersionnedXmlTest("eurostag-tutorial-example1-with-terminalMock-ext.xml", IidmXmlVersion.V_1_0); // TODO add versions mechanism for extensions
+        roundTripVersionnedXmlTest("eurostag-tutorial-example1-with-terminalMock-ext.xml", IidmXmlVersion.V_1_0);
         roundTripXmlTest(NetworkXml.read(getClass().getResourceAsStream(getVersionDir(IidmXmlVersion.V_1_0)
                         + "eurostag-tutorial-example1-with-terminalMock-ext.xml")),
                 NetworkXml::writeAndValidate,
                 NetworkXml::validateAndRead,
                 getVersionDir(IidmXmlConstants.CURRENT_IIDM_XML_VERSION) + "eurostag-tutorial-example1-with-terminalMock-ext.xml");
     }
 
+    @Test
+    public void testNotLatestVersionTerminalExtension() throws IOException {
+        Network network = NetworkXml.read(getClass().getResourceAsStream("/V1_1/eurostag-tutorial-example1-with-loadMockExt-1_2.xml"));
+        MemDataSource dataSource = new MemDataSource();
+        Properties properties = new Properties();
+        properties.put("iidm.export.xml.version", "1.1");
+        new XMLExporter().export(network, properties, dataSource);
+        try (InputStream is = new ByteArrayInputStream(dataSource.getData(null, "xiidm"))) {
+            assertNotNull(is);
+            compareXml(getClass().getResourceAsStream(getVersionDir(IidmXmlVersion.V_1_1) + "eurostag-tutorial-example1-with-loadMockExt-1_1.xml"),
+                    is);
+        }
+    }
+
     @Test
     public void testThrowErrorUncompatibleExtensionVersion() {
```

Comment:
Uncompatible should be Incompatible

Reply:
What is the difference between the two resource files in this test? Why do you use MemDataSource?

Reply:
> Why didn't you use the same name basis as the two other tests below?

For this test specifically, the version of the extension is supported, just not compatible with the given network version.

> What is the difference between the two resource files in this test? Why do you use MemDataSource?

I'm assuming you are talking about the test above. The difference between the two resource files is the version of the `loadMock` extension (1.2 for the imported file and 1.1 for the reference of the exported file). I am using `MemDataSource` because when exporting, I don't want the file to be physically on my machine but to be in memory (which `MemDataSource` allows to do).

Reply:
Comments have been added in tests to explain more in detail what is done